python/ray/serve/_private/application_state.py

```diff
@@ -413,6 +414,13 @@ def _start_or_reconcile_build_app_task(self) -> Tuple[Tuple, BuildAppStatus, str
         if config_version == self._target_state.code_version:
             return None, BuildAppStatus.NO_TASK_STARTED, ""
 
+        if (
+            self._build_app_task_info
+            and config_version == self._build_app_task_info.code_version
+            and self._build_app_task_info.finished
+        ):
+            return None, BuildAppStatus.NO_TASK_STARTED, ""
```

Comment:
this is just an internal status, right? it smells a little funny that we return "no_task_started" when in actuality it was started and then failed...

Reply:
Yup it's purely internal